packages/firebase_admob/README.md

```diff
@@ -182,6 +182,95 @@ RewardedVideoAd.instance.listener =
 Because `RewardedVideoAd` is a singleton object, it does not offer a `dispose`
 method.
 
+## Using native ads
+
+Native ads are presented to users via UI components that
+are native to the platform. (e.g. A
+[View](https://developer.android.com/reference/android/view/View) on Android or a
+[UIView](https://developer.apple.com/documentation/uikit/uiview?language=objc)
+on iOS). Using Flutter widgets to create native ads is NOT supported by
+this.
+
+### Android
+The Android Admob Plugin requires a class that implements `NativeAdFactory` which calls `createNativeAd(
+[UnifiedNativeAd](https://developers.google.com/android/reference/com/google/android/gms/ads/formats/UnifiedNativeAd) nativeAd,
+Map<String, Options> customOptions)` and returns a
+[UnifiedNativeAdView](https://developers.google.com/android/reference/com/google/android/gms/ads/formats/UnifiedNativeAdView).
+
+You can implement this in your `MainActivity.java` or create a separate class in the same directory
+as `MainActivity.java` as seen below:
+
+```java
+package my.app.path;
+
+import com.google.android.gms.ads.formats.UnifiedNativeAd;
+import com.google.android.gms.ads.formats.UnifiedNativeAdView;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin.NativeAdFactory;
+import java.util.Map;
+
+class NativeAdFactoryExample implements NativeAdFactory {
+  @Override
+  public UnifiedNativeAdView createNativeAd(
+      UnifiedNativeAd nativeAd, Map<String, Object> customOptions) {
```

Comment:
What's your thoughts on allowing multiple factories to be registered with a name (key). And then on the Dart side users can specify the factory to use for a given ad?

While the same can be achieved through customOptions (essentially passing a factory "key" as an option and changing the rendering logic accordingly) I wonder if it will be more convenient.

Another scenario to think about is having 2 packages that depend on the admob plugin being used by the same app, if we only allow a single factory their factories will conflict and override each other.

Reply:
For a plugin to set a `NativeAdFactory`, it would need access to the `FlutterEngine.getPlugins()`. This access is now deprecated for `FlutterPluginBinding`. 

A Flutter user would have to set this in their `MainActivity.configureFlutterEngine`, so I can only think of a scenario where two separate plugins have a public Java `NativeAdFactory`. Then, the user would register both in `MainActivity`. Do you think this would be common enough to need to register based on key?

Reply:
f2f discussion: we believe cross-plugin interaction on the platform side is useful, and we will want to add it on the mid-term. It doesn't seem to cost much to design this to be ready for that day, so we might as well do it now and prevent a future public API change.